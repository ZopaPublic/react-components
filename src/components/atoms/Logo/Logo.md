### Summary

Use it to render Zopa's logo as an SVG.

### Examples

- By default grows as wide as its container allows to:

```tsx
import { Logo, SizedContainer } from '@zopauk/react-components';

<SizedContainer size="short">
  <Logo />
</SizedContainer>;
```

- With custom size:

```tsx
import { Logo } from '@zopauk/react-components';

<Logo width="100px" />;
```

- Negative version:

```tsx { "props": { "style": { "backgroundColor": "black", "border": "none" } } }
import { Logo } from '@zopauk/react-components';

<Logo width="100px" negative />;
```
