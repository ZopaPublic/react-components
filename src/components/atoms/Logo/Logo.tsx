import React from 'react';
import { colors } from '../../../constants/colors';

export interface LogoProps extends React.SVGProps<SVGSVGElement> {
  width?: string;
  height?: string;
  negative?: boolean;
}

const Logo: React.FC<LogoProps> = ({ negative, ...rest }) => (
  <svg
    viewBox="0 0 104 19"
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
    role="img"
    aria-label="zopa logo"
    {...rest}
  >
    <title>Zopa Logo</title>
    <g>
      <path
        fill={negative ? colors.white : colors.brand}
        d="M36.90752,0 C42.6769778,0 46.8674844,3.93460861 46.8674844,9.36414355 C46.8674844,14.8199217 42.6769778,18.83326 36.90752,18.83326 C31.1380622,18.83326 26.9475556,14.8461649 26.9475556,9.36414355 C26.9475556,3.90836539 31.1380622,0 36.90752,0 Z M36.9851733,13.7701929 C39.0550044,13.7701929 40.7106844,11.9078615 40.7106844,9.38944951 C40.7106844,6.87197483 39.0550044,5.06212985 36.9851733,5.06212985 C34.8894578,5.06212985 33.1043556,6.87197483 33.1043556,9.38944951 C33.1043556,11.9078615 34.9153422,13.7701929 36.9851733,13.7701929 Z M16.4541867,1.04879158 C16.4541867,6.67702519 8.45959111,11.8600613 8.45959111,13.9529582 L16.1435733,13.9529582 L16.1435733,18.5708278 L0,18.5708278 L0,17.7048015 C0,12.0765679 7.99367111,6.89353176 7.99367111,4.8006349 L0.309688889,4.8006349 L0.309688889,0.183702546 L16.4541867,0.183702546 L16.4541867,1.04879158 Z M66.7153067,0.183702546 C71.4244267,0.183702546 74.1404444,2.56996113 74.1404444,6.76700194 C74.1404444,11.2255377 71.4244267,13.7964361 66.7153067,13.7964361 L64.2063644,13.7964361 L64.2063644,18.5708278 L58.5136356,18.5708278 L58.5136356,0.183702546 L66.7153067,0.183702546 Z M66.4046933,9.180441 C67.6989156,9.180441 68.44864,8.36690115 68.44864,6.95070449 C68.44864,5.58699426 67.6989156,4.79969764 66.4046933,4.79969764 L64.2063644,4.79969764 L64.2063644,9.180441 L66.4046933,9.180441 Z M96.6257067,0.612874778 L104,19 L97.6093156,19 L96.70336,16.4028584 L89.8994489,16.4028584 L88.9685333,19 L82.8108089,19 L90.4689067,0.612874778 L96.6257067,0.612874778 Z M91.4515911,11.619186 L95.2288711,11.619186 L93.3661156,6.24213753 L91.4515911,11.619186 Z"
      />
    </g>
  </svg>
);

Logo.defaultProps = {
  negative: false,
  height: '100%',
  width: '100%',
};

export default Logo;
