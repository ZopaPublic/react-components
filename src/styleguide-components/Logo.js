import React from 'react';
import styled from 'styled-components';
import zLogo from './svg/zopaLogo.svg';

const Container = styled.div`
  width: 170px;
  margin: 0 auto;
  text-align: center;
`;

const LogoTitle = styled.h1`
  font-size: 18px;
  font-weight: normal;
  color: #00b9a7;
  div {
    text-align: center;
    height: 50px;
    width: 170px;
    margin: 0 auto;
    &:hover rect {
      -webkit-animation: 0.5s draw linear forwards;
      animation: 0.5s draw linear forwards;
    }
    @keyframes draw {
      0% {
        stroke-dasharray: 140, 540;
        stroke-dashoffset: -474;
        stroke-width: 8px;
      }
      100% {
        stroke-dasharray: 760;
        stroke-dashoffset: 0;
        stroke-width: 2px;
      }
    }
  }
`;

const LinkRepo = styled.a`
  font-size: 14px;
  color: #c0e1d7;
  text-align: center;
  cursor: pointer;
  text-decoration: none;
`;

const LogoRect = styled.rect`
  fill: transparent;
  stroke-dasharray: 126px, 260px;
  stroke-dashoffset: -243px;
  stroke-width: 8px;
  stroke: #19f6e8;
`;

const ZLogo = styled.img`
  position: relative;
  top: -40px;
`;

export function LogoRenderer({ children }) {
  return (
    <Container>
      <LogoTitle>
        <div>
          <svg height="50" width="100%">
            <LogoRect height="50" width="100%" />
          </svg>
          <ZLogo src={zLogo} alt="Zopa logo" />
        </div>
        <LinkRepo
          href="https://github.com/zopaUK/react-components"
          title="Go to github repository"
          target="_blank"
          rel="noopener noreferrer"
        >
          {children}
        </LinkRepo>
      </LogoTitle>
    </Container>
  );
}

export default LogoRenderer;
