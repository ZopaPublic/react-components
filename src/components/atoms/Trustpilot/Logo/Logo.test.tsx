import axe from '../../../../../axe-helper';
import React from 'react';
import { render } from '@testing-library/react';
import Logo from './Logo';
import 'jest-styled-components';

describe('<TrustpilotLogo />', () => {
  it('renders without a11y violations', async () => {
    const { container } = render(<Logo />);
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });
});
