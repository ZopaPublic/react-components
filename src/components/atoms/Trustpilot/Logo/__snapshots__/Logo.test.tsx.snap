// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrustpilotLogo /> renders without a11y violations 1`] = `
<div>
  <svg
    aria-labelledby="trustpilot-logo-title trustpilot-logo-desc"
    height="16px"
    viewBox="0 0 212 53"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="trustpilot-logo-title"
    >
      Trustpilot Logo
    </title>
    <desc
      id="trustpilot-logo-desc"
    >
      An illustration of the Trustpilot Logo
    </desc>
    <path
      d="M55.7 18.7h21.5v4.1h-8.4v22.9h-4.6V22.8h-8.4zm20.5 7.5h4V30h.1c.1-.6.4-1.1.7-1.5l1.3-1.4c.5-.4 1.1-.8 1.7-1s1.2-.4 1.9-.4h1l.6.1v4.1l-.9-.1-.9-.1c-.7 0-1.4.2-2 .4s-1.2.7-1.6 1.3c-.5.6-.8 1.3-1.1 2.1s-.4 1.8-.4 2.9v9.3h-4.2zm30.7 19.4h-4.2v-2.7h-.1c-.5 1-1.3 1.8-2.3 2.4s-2.1.9-3.1.9c-2.5 0-4.4-.6-5.5-1.9S90 41.1 90 38.5V26.2h4.2v11.9c0 1.7.3 2.9 1 3.6.6.7 1.6 1.1 2.7 1.1.9 0 1.6-.1 2.2-.4s1.1-.6 1.4-1.1c.4-.5.6-1 .8-1.7.2-.6.2-1.3.2-2.1V26.2h4.2v19.4zm7.2-6.2c.1 1.3.6 2.1 1.4 2.6s1.8.8 2.9.8l1.3-.1 1.4-.4c.4-.2.8-.4 1.1-.8.3-.3.4-.8.4-1.3s-.2-1-.6-1.3c-.4-.4-.8-.6-1.4-.9l-2-.6-2.3-.5-2.3-.6c-.7-.2-1.4-.6-2-1s-1.1-1-1.4-1.6c-.4-.7-.5-1.5-.5-2.5 0-1.1.3-1.9.8-2.7.5-.7 1.2-1.3 1.9-1.7a12.4 12.4 0 018.1-.9c.9.2 1.7.6 2.4 1a6.2 6.2 0 012.7 4.7h-4c-.2-1-.7-1.7-1.4-2.1s-1.6-.5-2.5-.5l-1.1.1-1.2.3c-.3.1-.7.3-.9.6s-.4.6-.4 1c0 .5.2 1 .5 1.3.4.3.8.6 1.4.8l2 .5 2.3.5 2.3.6c.7.2 1.4.6 2 1s1 1 1.4 1.6.5 1.5.5 2.4c0 1.2-.3 2.1-.8 2.9s-1.2 1.5-2 2c-.9.5-1.8.9-2.8 1.1a15.3 15.3 0 01-6.4-.1c-1-.3-1.9-.7-2.7-1.3-.7-.6-1.4-1.3-1.8-2.1s-.7-1.8-.7-3zm14-13.2h3.2v-5.9h4.2v5.9h3.8v3.2h-3.8v10.4l.1 1.2c0 .3.1.6.3.8.1.2.3.4.6.5l1.1.2h.9l.9-.1v3.3l-1.4.2-1.4.1c-1.1 0-2-.1-2.7-.3s-1.2-.5-1.6-1c-.4-.4-.6-.9-.8-1.6a15 15 0 01-.2-2.1V29.4h-3.2zm14.2 0h4v2.6h.1c.6-1.1 1.4-1.9 2.5-2.4s2.2-.7 3.5-.7c1.5 0 2.8.3 3.9.8s2.1 1.3 2.8 2.2c.7 1 1.3 2.1 1.7 3.3a13.4 13.4 0 01.1 7.8c-.3 1.2-.8 2.3-1.5 3.3s-1.5 1.7-2.6 2.3-2.3.9-3.7.9l-1.9-.2-1.8-.6c-.6-.2-1.1-.6-1.6-1s-.9-.9-1.2-1.4h-.1v9.7h-4.2zm14.8 9.7c0-.9-.1-1.7-.3-2.6-.2-.8-.5-1.5-1-2.2-.4-.6-1-1.2-1.7-1.5-.7-.4-1.5-.6-2.3-.6-1.8 0-3.1.6-4 1.9s-1.3 2.9-1.3 5c0 1 .1 1.9.4 2.7s.6 1.6 1.1 2.2a4 4 0 001.7 1.4c.7.4 1.4.5 2.3.5 1 0 1.8-.2 2.5-.6s1.2-.9 1.6-1.6c.4-.6.7-1.4.9-2.2l.1-2.4zm7.5-17.2h4.2v4.1h-4.2zm0 7.5h4.2v19.5h-4.2V26.2zm8-7.5h4.2v26.9h-4.2zm17.2 27.5c-1.5 0-2.9-.3-4.1-.8s-2.2-1.2-3.1-2.1c-.8-.9-1.5-2-1.9-3.2-.4-1.3-.7-2.6-.7-4.1a11 11 0 012.6-7.3c.8-.9 1.9-1.6 3.1-2.1s2.6-.8 4.1-.8 2.9.3 4.1.8 2.2 1.2 3.1 2.1c.8.9 1.5 2 1.9 3.2.4 1.3.7 2.6.7 4.1s-.2 2.9-.7 4.1a8.5 8.5 0 01-5 5.3c-1.2.5-2.6.8-4.1.8zm0-3.4c.9 0 1.8-.2 2.4-.6.7-.4 1.3-.9 1.7-1.6.4-.6.8-1.4 1-2.2a10.7 10.7 0 000-5 6.2 6.2 0 00-2.7-3.8c-.7-.4-1.5-.6-2.5-.6-.9 0-1.8.2-2.5.6s-1.3.9-1.7 1.6c-.4.6-.8 1.4-1 2.2a10 10 0 000 5c.2.8.5 1.6 1 2.2.4.6 1 1.2 1.7 1.6.8.4 1.7.6 2.6.6zm10.9-16.6h3.2v-5.9h4.2v5.9h3.8v3.2h-3.8v10.4l.1 1.2c0 .3.1.6.3.8.1.2.3.4.6.5l1.1.2h.9l.9-.1v3.3l-1.4.2-1.4.1c-1.1 0-2-.1-2.7-.3s-1.2-.5-1.6-1c-.4-.4-.6-.9-.8-1.6a15 15 0 01-.2-2.1V29.4h-3.2z"
    />
    <path
      d="M50.8 18.7H31.4L25.4 0l-6 18.7H0l15.7 11.6-6 18.7 15.7-11.6L41.1 49l-6-18.7z"
      fill="#00B679"
    />
    <path
      d="M36.4 34.5l-1.3-4.2-9.7 7.1z"
      fill="#02917c"
    />
  </svg>
</div>
`;
